colcon_core/task/python/build.py

```diff
@@ -105,9 +105,15 @@ def __init__(self):  # noqa: D107
             # to maintain the desired order
             if additional_hooks is None:
                 additional_hooks = []
+            base_path = args.build_base
+            # if the Python packages are in a subdirectory
+            # that needs to be appended to the build directory
+            package_dir = setup_py_data.get('package_dir') or {}
+            if '' in package_dir:
```

Comment:
> It might be worth warning if something other than `''` is in `package_dir`.

Not sure if a warning would be helpful if the result is broken afterwards anyway... Maybe falling back to not symlinking in this car would be better? Anyway I will keep that for a separate PR.